src/ga/ga_core.rs

```diff
@@ -27,24 +30,24 @@ impl Default for GAFlags
 pub trait GAIndividual
 {
     // Instance
-    fn crossover(&self, other : &Self) -> Box<Self>;
-    fn mutate(&mut self, pMutation : f32);
-    fn evaluate(&mut self) -> f32;
+    fn crossover(&self, other: &Self, rng_ctx: &mut GARandomCtx) -> Box<Self>;
+    fn mutate(&mut self, pMutation: f32, rng_ctx: &mut GARandomCtx);
+    fn evaluate(&mut self, evaluation_ctx: &mut Any);
```

Comment:
I ended up leaving it, and allowing for a context to be passed, I think this is the right way :)
@carlos-lopez-garces I agree with your point of the evaulation and score ownership, so this should make you happier :)